### PR TITLE
Update benchmark runner to support hydra 1.1 and distinguish threaded and unthreaded C runtime.

### PR DESCRIPTION
**Pull request by cmnrd**
This PR does 2 things:
- Updating the hydra dependency. A new version of hydra (1.1) has been released. Hydra 1.1  brings a few important changes and also deprecates several functions. This PR adjusts both our configs and the runner script to the new requirements of hydra 1.1. This also means, that older hydra versions (<1.1) are not supported anymore.
- Adding a new target: the unthreaded C runtime. With this change, we can use `target=lf-c` for threaded C programs, and `target=lf-c-unthreaded` for unthreaded ones. Note, that this requires appropriate cog code in the individual C benchmarks to work correctly.

This PR is based on #406 and should only be merged after merging #406.

### CODE REVIEW

**Review by Feliix42 — APPROVED** [review]
LGTM

File: benchmark/C/Savina/src/micro/Counting.lf [hunk 1]
```diff
@@ -38,7 +38,16 @@
  ]]] */
  // [[[end]]]
 
-target C;
+target C {
+    /* [[[cog
+      if (threaded_runtime=="True"):
+          cog.outl(f"threads: {threads},")
```
Comment (lhstrh) [inline benchmark/C/Savina/src/micro/Counting.lf:44]: Should we take this PR as an opportunity to address https://github.com/icyphy/lingua-franca/issues/290?

**Review by lhstrh — APPROVED** [review]
Looks good, but I'm thinking we might want to address https://github.com/icyphy/lingua-franca/issues/290 in this PR as well, to avoid breaking this functionality by the time we get around to implementing the change proposed in that issue.

File: benchmark/C/Savina/src/micro/Counting.lf (hunk 1 above)
Comment (cmnrd) [reply benchmark/C/Savina/src/micro/Counting.lf:44]: I think #290 is a bigger change and I would rather address this separately

Comment (lhstrh) [reply benchmark/C/Savina/src/micro/Counting.lf:44]: OK, fair enough!

**Review by mattchorlian — APPROVED** [review]
Looks good to me as well! Will all the config files need an additional lf-c-unthreaded configuration? And if so, should we address that here or in another PR?

**Comment by cmnrd** [comment]
No, the benchmark configs don't need to change. All the magic happens in the target configurations (and partly in the runner script). The benchmarks only need to contain the appropriate cog code. This we should also adjust in #445 and #481 once this is merged. BTW, I will have a closer look at those PRs next week, so that we can ready them up for merging.

**Pull request merged**